courtlistener/csv_to_dolma.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+"""
+Created by zhenlinx on 01/19/2024
+"""
+import argparse
+import csv
+import logging
+import os
+import sys
+from datetime import datetime
+
+from licensed_pile.licenses import PermissiveLicenses
+from licensed_pile.logs import configure_logging
+from licensed_pile.write import to_dolma
+
+SOURCE_NAME = "CourtListenerOpinion"
+
+csv.field_size_limit(sys.maxsize)
+
+configure_logging("court-listener-opinion")
+
+
+def make_record_generator(file_path):
+    with open(file_path, "r") as csvfile:
+        # Create a CSV reader object
+        reader = csv.DictReader(csvfile)
+
+        # Yield a dictionary for each row
+        for row in reader:
+            # 'row' is a dictionary with column headers as keys
+
+            if not row["plain_text"]:
+                pass  # TODO load from row["download_url"] if not null
+            else:
+                yield {
+                    "id": row["id"],
+                    "text": row["plain_text"],
+                    "source": SOURCE_NAME,
+                    "added": datetime.utcnow().isoformat(),
+                    "created": row["data_created"],
+                    "metadata": {
+                        "license": str(PermissiveLicenses.PD),
+                        "url": row["download_url"],
+                    },
+                }
+
+
+def main(args):
+    example_generator = make_record_generator(args.input_file)
+    output_file_base_name = os.path.basename(args.input_file).replace(
+        ".csv", ".jsonl.gz"
+    )
+    to_dolma(example_generator, args.output_dir, output_file_base_name, args.shard_size)
+    logging.info(f"Saved {args.input_file} as dolma shared files at {args.output_dir}")
```

Comment:
With the new logger you need to do this:

```python
logger = logs.get_logger("court-listener-opinion")
logger.info(...)
```

instead of using the root logger (`logging.info(...)`)

Reply:
Can I do the following?
```
logger = configure_logging("court-listener-opinion")

...

logger.info(...)
```